### DISCUSSION TITLE
return of `Any` being reported as possibly unbound (via VS Code's pylance)

### DISCUSSION THREAD

**Discussion by jecluis** [microsoft/pyright#1362] [Q&A] [answered]
Hi all,

I've got this slightly annoying issue where a function returning `Any` keeps having its return marked as possibly unbound. I am importing `Any` from `typing`, but still no joy. `mypy`, for instance, does not report anything of the such; `pylance` does.

I've combed through the internet, and found nothing pertaining this, which means one of two things: either I'm doing something incredibly wrong here, or I've hit just the sweet spot for some corner case. I'm leaning towards the former, but would appreciate getting some confirmation.

Cheers

**Comment by erictraut** [comment]
I'll need to see your code or (preferably) a minimal code snippet that exhibits the problem you're seeing.

To my knowledge, mypy doesn't track bound vs unbound variables.

I'm not sure what you mean by "returning `Any`". `Any` is a type that is typically used in type annotations. It's not a value that you would ever "return". Are you saying that you are returning a value that has the type `Any`?

**Reply by jecluis** [reply]
Hi Eric.

I'm very sorry for not being clearer. I meant a return annotation for type `Any`. See below.

![image](https://user-images.githubusercontent.com/506918/104291749-74c2db00-54ab-11eb-9fdc-d820cc100e55.png)

Does this make sense? Do you need anything else?

**Comment by erictraut** [comment, marked as answer]
Annotating a return type with `Any` quite common, so that in itself is not unusual. There must be something else about your configuration or code file that is resulting in this problem.

I suspect it's a problem in the way you are importing `Any. Perhaps you are conditionally importing that symbol? Something like this:
```python
test: bool = False

if test:
    from typing import Any

def foo() -> Any:
    return 123
```

A few follow-up questions:
1. Where are you importing `Any` from?
2. Is the import statement at the top level of the module or within an if/else or try/except statement?
3. If you copy the `foo` function into its own ".py" file (and add an import statement), do you still see the problem?
4. Is the symbol `Any` used in any other manner within your source file? For example `del Any`?

**Reply by jecluis** [reply]
> I suspect it's a problem in the way you are importing `Any. Perhaps you are conditionally importing that symbol? Something like this:

_sigh_

You are absolutely correct. The import is wrapped in a `try/except` block for reasons that escape me (guessing backwards compatibility); taking it out of the block makes it work.

Thank you, and very sorry for the noise.
